### PR TITLE
play 2.4 module definition

### PR DESCRIPTION
**Pull request by pbelmann**
see #43 

**Summary**:

* Play Module implementation https://www.playframework.com/documentation/2.4.x/PluginsToModules

* Removal of Play Plugin implementation as suggested in https://www.playframework.com/documentation/2.4.x/PluginsToModules

* Major version incremented due to the removal of the plugin implementation

**Usage**:

* PlayJongo must be injected:

Example:

~~~Java
public class User {

    public static PlayJongo jongo = Play.application().injector().instanceOf(PlayJongo.class);
	
    public static MongoCollection users() {
        return jongo.getCollection("Users");
    }
}
~~~

**TODO**:

If this PR is merged then the readme file must be updated.

### CODE REVIEW

File: app/uk/co/panaxiom/playjongo/IPlayJongo.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package uk.co.panaxiom.playjongo;
+
+public interface IPlayJongo {
```
Comment (magro) [inline app/uk/co/panaxiom/playjongo/IPlayJongo.java]: What is this interface good for? Why not just bind PlayJongo.class directly?

**Comment by magro** [comment]
Nice, LGTM! Can you update the README as part of this PR?

**Comment by pbelmann** [comment]
@magro I removed the unnecessary interface and updated the README. 
You should update the repository description. It still states `Play 2.2 Module for Jongo http://jongo.org/`

**Comment by magro** [comment]
@pbelmann Great, looks good! Can you please squash commits to a single one? Thanks!

For the repository description I'm missing permissions as it seems.
@alexanderjarvis Can you update the repository description (or give me the required permissions)? It still states "Play 2.2 Module for Jongo http://jongo.org/".

**Comment by pbelmann** [comment]
@magro  I squashed my commits into two because I rebased my branch before updating the readme. I hope it's ok.
The history of this PR shows me 5 commits but there are actaully 2.
See here: 
https://github.com/pbelmann/play-jongo/commits/feature/play-2.4-module

**Comment by magro** [comment]
Thanks! 1.0.0-jongo1.2 is on its way to maven central.

**Comment by pbelmann** [comment]
Great! thanks!

**Comment by magro** [comment]
@pbelmann There was an issue with the `bind(PlayJongo.class).to(PlayJongo.class)`, I changed this to `bind(PlayJongo.class).toSelf)`. This is contained in 1.0.1-jongo1.2 I just released.

**Comment by pbelmann** [comment]
Have not noticed this bug in my play project. Anyway thanks for the fix.

**Pull request merged**